public/app/plugins/datasource/prometheus/workers/StreamJSONResponse.worker.ts

```diff
@@ -0,0 +1,95 @@
+import oboe from 'oboe';
+import { JSON_STREAM_DONE } from './consts';
+
+// See: https://github.com/microsoft/TypeScript/issues/20595#issuecomment-587297818
+const postMessage = ((self as unknown) as Worker).postMessage;
+let isFetching = false;
+
+export type StreamJSONResponsePayload = {
+  data: {
+    url: string;
+    chunkSize?: number;
+    hasObjectResponse?: boolean;
+    headers?: any;
+    limit?: number;
+    path?: string;
+    withCredentials?: boolean;
+  };
+};
+
+export interface StreamJSONResponseWorker extends Worker {
+  postMessage(message: StreamJSONResponsePayload['data'], transfer: Transferable[]): void;
+  postMessage(message: StreamJSONResponsePayload['data'], options?: PostMessageOptions): void;
+}
+
+export function streamJSONResponse(data: StreamJSONResponsePayload['data'], callback: (arg: any) => void) {
```

Comment:
Nitpick: Maybe we can change to `callback: (arg: JSON_STREAM_DONE | any)` would not be much of help with type checking but would add some context for reader.

Reply:
Good catch! I've improved on it further by defining types for `ArrayChunk` and `ObjectChunk`.